chrome/content/zotero/integration/quickFormat.js

```diff
@@ -79,15 +88,46 @@ var Zotero_QuickFormat = new function () {
 			
 			new WindowDraggingElement(document.querySelector("window.citation-dialog"), window);
 			
-			qfs = document.querySelector(".citation-dialog.search");
-			qfi = document.querySelector(".citation-dialog.iframe");
 			qfb = document.querySelector(".citation-dialog.entry");
+			qfe = document.querySelector(".citation-dialog.editor");
+			qfb.addEventListener("click", _onQuickSearchClick, false);
+			qfb.addEventListener("keypress", _onQuickSearchKeyPress, false);
+			qfe.addEventListener("dragover", _onBubbleDragOver);
+			qfe.addEventListener("drop", _onBubbleDrop, false);
 			qfbHeight = qfb.scrollHeight;
 			referencePanel = document.querySelector(".citation-dialog.reference-panel");
 			referenceBox = document.querySelector(".citation-dialog.reference-list");
-			
+			// Navigation within the reference panel
+			referenceBox.addEventListener("keypress", (event) => {
+				// Enter or ; selects the reference
+				if (event.key == "Enter" || event.charCode == 59) {
+					event.preventDefault();
+					event.stopPropagation();
+					event.target.closest("richlistitem").click();
+				}
+				// Tab will move focus back to the input field
+				else if (event.key === "Tab") {
+					event.preventDefault();
+					event.stopPropagation();
+					_lastFocusedInput.focus();
+				}
+				// Can keep typing from the reference box
+				else if (event.key.length === 1 || event.key === 'Backspace') {
+					event.preventDefault();
+					if (event.key === 'Backspace') {
+						_lastFocusedInput.value = _lastFocusedInput.value.slice(0, -1);
+					}
+					else {
+						_lastFocusedInput.value += event.key;
+					}
+					_lastFocusedInput.dispatchEvent(new Event('input', { bubbles: true }));
+					_lastFocusedInput.focus();
+				}
+				else if (["ArrowDown", "ArrowUp"].includes(event.key)) {
+					handleItemSelection(event);
+				}
+			});
 			if (Zotero.isWin) {
-				referencePanel.style.marginTop = "-29px";
 				if (Zotero.Prefs.get('integration.keepAddCitationDialogRaised')) {
```

Comment:
Thank you for pointing that out - I must have removed it while getting rid of the iframe because it had no effect afterwards. 

The reference panel, however, was indeed showing too far below the dialog when the editor had <3 lines. After some experimentation, adding `marginBottom`  to the `dialog` (aka `qfb`), similar to macOS, seems to have solved it.

In regards to the reason as to why this is all needed, we can't seem to set the height of the window below a specific value (`window.resizeTo` with a height of, say, 50 would set the height to ~80 for me). Likely something to do with counting the height of the title? That extra window height is what would create the gap, and I had to set the negative margin on the dialog to undo it